contracts/paratii/TcrPlaceholder.sol

```diff
@@ -0,0 +1,169 @@
+pragma solidity ^0.4.18;
+
+import "./ParatiiToken.sol";
+
+/// WARNING , NOT PRODUCTION READY, based on ADCHAIN Registry.
+/// THIS IS JUST A PLACEHOLDER..
+contract TcrPlaceholder {
+
+  // Events
+  // ---------
+
+  event _Application(string videoId, uint deposit);
+  event _NewVideoWhitelisted(string videoId);
+
+  struct Listing {
+    uint applicationExpiry; // Expiration date of apply stage
+    bool whitelisted;       // registration status
+    address owner;          // Owner of the video.
+    uint unstakedDeposit;   // unlocked tokens
+    uint challengeID;       // ID of canonical challenge
+  }
+
+  // Maps of videoId Hashes to listing data.
+  mapping(bytes32 => Listing) public listings;
+
+  ParatiiToken public token;
+  uint public minDeposit;
+  uint public applyStageLen;
+
+  // ---------------------------------------------------------------------------
+  // CONSTRUCTOR
+  // ------------------
+
+  function TcrPlaceholder(
+    address _tokenAddr,
+    uint _minDeposit,
+    uint _applyStageLen
+  ) {
+    token = ParatiiToken(_tokenAddr);
+    minDeposit = _minDeposit;
+    applyStageLen = _applyStageLen; // 100 blocks.
+  }
+
+  // ---------------------------------------------------------------------------
+  // Video creator functions
+  //------------------------
+
+  /**
+   * allow video creator to apply for whitelisting for a video.
+   * @param  _videoId  the videoId
+   * @param  _amount   the amount to Stake
+   */
+  function apply(string _videoId, uint _amount) external {
+    require(!isWhitelisted(_videoId));
+    require(!appWasMade(_videoId));
+    require(_amount >= minDeposit);
+
+    // set owner
+    Listing storage listing = listings[keccak256(_videoId)];
+    listing.owner = msg.sender;
+
+    // transfer tokens from user to TcrPlaceholder contract
+    require(token.transferFrom(listing.owner, this, _amount));
```

Comment:
I don't _think_ you need the require as the transfer throws an error if it does not succeed

Reply:
I'd rather have require just to make sure this returns `true`

Reply:
+1